target/linux/generic/hack-6.6/765-mxl-gpy-control-LED-reg-from-DT.patch

```diff
@@ -55,7 +55,19 @@ Signed-off-by: David Bauer <mail@david-bauer.net>
  #define PHY_PMA_MGBT_POLARITY	0x82
  #define PHY_MDI_MDI_X_MASK	GENMASK(1, 0)
  #define PHY_MDI_MDI_X_NORMAL	0x3
-@@ -260,6 +267,35 @@ out:
+@@ -228,7 +235,10 @@ static int gpy_ack_interrupt(struct phy_
+ 
+ 	/* Clear all pending interrupts */
+ 	ret = phy_read(phydev, PHY_ISTAT);
+-	return ret < 0 ? ret : 0;
++	if (ret < 0)
++		return ret;
++
++	eturn gpy_led_write(phydev);
```

Comment:
```suggestion
+	return gpy_led_write(phydev);
```

Reply:
I made the proposed modification but building still fails.
```
...
make -C /scratch/union/build_dir/target-aarch64_cortex-a53_musl/linux-mediatek_filogic/linux-6.6.36 KCFLAGS="-fmacro-prefix-map=/scratch/union/build_dir/target-aarch64_cortex-a53_musl=target-aarch64_cortex-a53_musl -mcpu=cortex-a53 -mcpu=cortex-a53" HOSTCFLAGS="-O2 -I/scratch/union/staging_dir/host/include  -Wall -Wmissing-prototypes -Wstrict-prototypes" CROSS_COMPILE="aarch64-openwrt-linux-musl-" ARCH="arm64" KBUILD_HAVE_NLS=no KBUILD_BUILD_USER="" KBUILD_BUILD_HOST="" KBUILD_BUILD_TIMESTAMP="Thu Jun 27 14:11:11 2024" KBUILD_BUILD_VERSION="0" KBUILD_HOSTLDFLAGS="-L/scratch/union/staging_dir/host/lib" CONFIG_SHELL="bash" V=''  cmd_syscalls=  CC="aarch64-openwrt-linux-musl-gcc" KERNELRELEASE=6.6.36 Image dtbs modules
make[5]: Entering directory '/scratch/union/build_dir/target-aarch64_cortex-a53_musl/linux-mediatek_filogic/linux-6.6.36'
  CC      drivers/net/phy/mxl-gpy.o
drivers/net/phy/mxl-gpy.c: In function 'gpy_ack_interrupt':
drivers/net/phy/mxl-gpy.c:241:16: error: implicit declaration of function 'gpy_led_write' [-Werror=implicit-function-declaration]
  241 |         return gpy_led_write(phydev);
      |                ^~~~~~~~~~~~~
drivers/net/phy/mxl-gpy.c: At top level:
drivers/net/phy/mxl-gpy.c:283:12: error: static declaration of 'gpy_led_write' follows non-static declaration
  283 | static int gpy_led_write(struct phy_device *phydev)
      |            ^~~~~~~~~~~~~
drivers/net/phy/mxl-gpy.c:241:16: note: previous implicit declaration of 'gpy_led_write' with type 'int()'
  241 |         return gpy_led_write(phydev);
      |                ^~~~~~~~~~~~~
drivers/net/phy/mxl-gpy.c:283:12: error: 'gpy_led_write' defined but not used [-Werror=unused-function]
  283 | static int gpy_led_write(struct phy_device *phydev)
      |            ^~~~~~~~~~~~~
cc1: all warnings being treated as errors
make[10]: *** [scripts/Makefile.build:243: drivers/net/phy/mxl-gpy.o] Error 1
make[9]: *** [scripts/Makefile.build:480: drivers/net/phy] Error 2
make[8]: *** [scripts/Makefile.build:480: drivers/net] Error 2
make[7]: *** [scripts/Makefile.build:480: drivers] Error 2
make[6]: *** [/scratch/union/build_dir/target-aarch64_cortex-a53_musl/linux-mediatek_filogic/linux-6.6.36/Makefile:1921: .] Error 2
make[5]: *** [Makefile:234: __sub-make] Error 2
make[5]: Leaving directory '/scratch/union/build_dir/target-aarch64_cortex-a53_musl/linux-mediatek_filogic/linux-6.6.36'
make[4]: *** [Makefile:18: /scratch/union/build_dir/target-aarch64_cortex-a53_musl/linux-mediatek_filogic/linux-6.6.36/.modules] Error 2
make[4]: Leaving directory '/scratch/union/target/linux/mediatek'
make[3]: *** [Makefile:11: compile] Error 2
make[3]: Leaving directory '/scratch/union/target/linux'
time: target/linux/compile#3.51#2.40#6.24
    ERROR: target/linux failed to build.
make[2]: *** [target/Makefile:32: target/linux/compile] Error 1
make[2]: Leaving directory '/scratch/union'
make[1]: *** [target/Makefile:25: /scratch/union/staging_dir/target-aarch64_cortex-a53_musl/stamp/.target_compile] Error 2
make[1]: Leaving directory '/scratch/union'
make: *** [/scratch/union/include/toplevel.mk:248: world] Error 2
```

Reply:
66 -	return gpy_led_write(phydev);